### PR TITLE
Remove all dcgmproftester

### PR DESCRIPTION
**Pull request by gabrielcocenza**
`dcgmproftester` has diffetent versions and needs libcuda which is not currently shipped in the snap, so it does not work out of the box. Moreover, some versions like the `dcgmproftester10` is not present in the arm64 version that breaks the build.

This PR removes all dcgmproftester and libs that needs cuda until we have a more clear way on how to ship this service and keep the support for multi-arch

Closes: #18

### CODE REVIEW

**Comment by gabrielcocenza** [comment]
Just to complement here are the logs proving that removing the `dcgmproftester` will make the build work for arm64:
```
2024-09-13 15:23:11.269 :: 2024-09-13 15:17:41.824 - library: libnvml_injection.so.1: unused library 'usr/lib/aarch64-linux-gnu/libnvml_injection.so.1.0'. (https://snapcraft.io/docs/linters-library)     
2024-09-13 15:23:11.271 :: 2024-09-13 15:17:41.826 - library: libnvperf_host.so: unused library 'usr/lib/aarch64-linux-gnu/libnvperf_dcgm_host.so'. (https://snapcraft.io/docs/linters-library)            
2024-09-13 15:23:11.272 :: 2024-09-13 15:17:41.829 pack_snap: output='.', compression='xz'                                                                                                                 
2024-09-13 15:23:11.274 :: 2024-09-13 15:17:41.832 pack_snap: check skeleton                                                                                                                               
2024-09-13 15:23:11.276 :: 2024-09-13 15:17:42.297 Creating snap package...                                                                                                                                
2024-09-13 15:23:11.277 :: 2024-09-13 15:17:42.301 Pack command: ['snap', 'pack', '--filename', 'dcgm_3.3.7_arm64.snap', '--compression', 'xz', PosixPath('/root/prime'), PosixPath('/root/project')]      
2024-09-13 15:23:11.279 :: 2024-09-13 15:23:03.007 Packed dcgm_3.3.7_arm64.snap                                                                                                                            
2024-09-13 15:23:11.291 Executing on host: lxc --project snapcraft config device show local:snapcraft-dcgm-on-arm64-for-arm64-259410                                                                       
2024-09-13 15:23:12.647 Executing on host: lxc --project snapcraft config device remove local:snapcraft-dcgm-on-arm64-for-arm64-259410 disk-/root/project                                                  
2024-09-13 15:23:14.507 Executing on host: lxc --project snapcraft stop local:snapcraft-dcgm-on-arm64-for-arm64-259410 
```

Currently, the [build](https://github.com/canonical/dcgm-snap/actions/runs/10840428244/job/30082690448?pr=15) is failing

**Review by aieri — CHANGES_REQUESTED** [review]
Can you please also remove the executables and related libraries? They are the greatest contributors to the size of the snap so we have a chance to trim a lot of MBs. See #2 

Also please update the description in snapcraft.yaml to mention the absence of these tools as we are diverging from upstream. Of course we have the main documentation PR pending, so a quick sentence is enough to ensure we don't forget.

**Comment by gabrielcocenza** [comment]
Snap package now is only `25M` compared to `800M` that was before this change

**Review by samuelallan72 — DISMISSED** [review]

File: snap/local/scripts/remove_dcgmproftester.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+FILES_TO_REMOVE=(
+    "dcgmproftester*"
+    "libdcgm_cublas_proxy*"
+)
+
+
+echo "Removing dcgmproftester libs"
+
+
+for lib in "${FILES_TO_REMOVE[@]}"; do
```
Comment (aieri) [inline snap/local/scripts/remove_dcgmproftester.sh:13]: non blocking nit: `dcgmproftester*` are not libs

File: snap/local/scripts/remove_dcgmproftester.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    find "$SNAPCRAFT_PRIME" -type f -name "$lib" | while read -r file; do
+        echo "Removing file $file"
+        rm -f "$file"
+    done
+done
+
+
+echo "Removing libs that are missing cuda libs"
```
Comment (aieri) [inline snap/local/scripts/remove_dcgmproftester.sh:21]: non blocking nit: I don't understand this line, I suggest rephrasing

**Review by aieri — APPROVED** [review]
+1, with some minor suggestions

**Review by aieri — APPROVED** [review]

**Review by samuelallan72 — APPROVED** [review]

**Pull request merged**